custom/enikshay/private_sector_datamigration/management/commands/create_dtos_and_agencies.py

```diff
@@ -0,0 +1,102 @@
+from django.core.management import BaseCommand
+
+from corehq.apps.locations.models import SQLLocation, LocationType
+from corehq.apps.locations.tasks import make_location_user
+from custom.enikshay.private_sector_datamigration.models import Agency, UserDetail
+
+
+class Command(BaseCommand):
+
+    def add_arguments(self, parser):
+        parser.add_argument('domain')
+        parser.add_argument('state_code')
+        parser.add_argument('district_code')
+        parser.add_argument('parent_loc_id')
+        parser.add_argument('user_level', choices=['dev', 'real', 'test'])
+        parser.add_argument('org_ids', metavar='org_id', nargs='*', type=int)
+
+    def handle(self, domain, state_code, district_code, parent_loc_id, user_level, org_ids, **options):
+        dto_parent = SQLLocation.active_objects.get(location_id=parent_loc_id)
+        for org_id in org_ids:
+            dto = self.create_dto(domain, state_code, district_code, dto_parent, org_id)
+            for agency in self.get_agencies_by_state_district_org(state_code, district_code, org_id):
+                agency_loc = self.create_agency(domain, agency, dto, org_id)
+                self.create_user(agency_loc, user_level)
+
+    def create_dto(self, domain, state_code, district_code, dto_parent, org_id):
+        return SQLLocation.objects.create(
+            domain=domain,
+            name=self._get_org_name_by_id(org_id),
+            site_code='%s_%s_%d' % (state_code, district_code, org_id),
+            location_type=LocationType.objects.get(
+                domain=domain,
+                code='dto',
+            ),
```

Comment:
Not sure how many of these you'll be creating in one go, but if it's a lot, this could also be cached.